terraform/mesh_aws/mailboxes/ssm.tf

```diff
@@ -0,0 +1,87 @@
+resource "aws_ssm_parameter" "mailbox_allowed_senders" {
+  # SSM does not support empty parameters
+  count = var.mailbox.allowed_senders != null ? 1 : 0
+
+  name      = "/${var.name}/mesh/mailboxes/${var.mailbox_id}/ALLOWED_SENDERS"
```

Comment:
These should all be ${var.name_prefix} as we don't want {environment}-mesh/mesh

Reply:
We should not special case just this one use of the naming

The suffix "-mesh" was added as we have no control of the `name_prefix` that gets passed in, but we want to ensure that everything at least somewhere in it's name contains the word "mesh" without having to have that hard coded on every single terraform resource (and any future work also remembering to add it)

If we don't like the stutter here, I think we should remove the `/mesh/`, as we already know that the `name` will have `-mesh` as a suffix

Reply:
Ok, I think you're right on the tagging issue. It should be up to the project using this module. The same applies to the configuration of this module, then - it's not part of this module but part of the project. They also have different ways to configure it, they can for instance just configure the cert/url/etc params once for the whole environment, and then use tagging on objects, or configure completely different paths for buckets and keys for inboxes and outboxes that make sense to their application logic. Therefore they would do it under /environment/blah, and wouldn't normally do /environment-mesh/... and /environment-this/... and /environment-that/... but /evironment/this/... /environment/that/... and /environment/mesh/...  Well, I wouldn't because it makes no sense when your separators are '/', not '-'.
It's nice you give them an example, but you're forcing them into your chosen pattern, so this should be an exception to the 'rule'.

Reply:
Perhaps we shouldn't actually force this bit of terraform on the project, but instead give them it in the docs?

Reply:
Otherwise they need to create terraform to delete yours and rebuild this whole thing the way they want.

Reply:
The configuration of the module is it's contract with the consumer, I disagree that the ownership of configuration is the same as with the tags. 
The tags a project use are very specific to their use case and have no affect on how this module functions. Many resources in this module are based on being programmatically driven by its configuration contract, therefore it needs to be met by the consumer. 
Later iterations of the module could include functionality as you've described, like being able to pass in a "cert/url/etc params once for the whole environment" to overwriting the internal value, but that's outside of the current scope and requirements of this PR.

As this is just an internal implementation detail, changing of the code from: `name = "/${var.name}/mesh/mailboxes..."`
to: `name = "/${var.name_prefix}/mesh/mailboxes..."`
will make no difference to the functionality of this module, will that be sufficient to get you approval of this PR?

Reply:
You made me think about it, which was good. I was incorrect, we shouldn't be creating the parameter store config at all because we don't know what config mechanism the project wants to use as they have quite a few different options. Or create it but don't create the mapping part. Because they will have the option to automagically configure it by posting a folder as Steve W asked for:
/{bucket}/outbound/{src_mailbox}/{dest_mailbox}/{workflow_id}/{file}
They may also want different folders from the ones you create for them so they will have to remove them with their terraform.
If you want to I'll ok the merge and we'll have to rip the stuff out again later.

Reply:
In which case it doesn't matter what it does.